mixnet/client/src/sender.rs

```diff
@@ -109,13 +142,129 @@ impl<R: Rng> Sender<R> {
 
         let mu: std::sync::Arc<tokio::sync::Mutex<tokio::net::TcpStream>> =
             pool.get_or_init(&addr).await?;
+        let arc_socket = mu.clone();
         let mut socket = mu.lock().await;
-        let body = Body::new_sphinx(packet);
-        body.write(&mut *socket).await?;
+        let bytes = packet.to_bytes();
+        let packet_id = PacketId::from_bytes(&bytes);
+        match Body::write_sphinx_packet_bytes(&mut *socket, &bytes).await {
+            Ok(_) => {
+                tracing::info!("Sent a Sphinx packet successuflly to the node: {addr}");
+                Self::insert_packet_id(&ack_cache, addr, packet_id).await;
 
-        tracing::debug!("Sent a Sphinx packet successuflly to the node: {addr:?}");
+                tokio::spawn(async move {
+                    Self::retry_backoff(
+                        ack_cache,
+                        max_retries,
+                        retry_delay,
+                        packet_id,
+                        bytes,
+                        addr,
+                        arc_socket,
+                    )
+                    .await;
+                });
+                Ok(())
+            }
+            Err(e) => {
+                if let Some(err) = e.downcast_ref::<std::io::Error>() {
+                    match err.kind() {
```

Comment:
Not to be done in this PR. But we have probably reached the point in where `DynError` is too generic. We probably need to start composing exact error with **`thiserror`**